src/app/recipient/components/recipient-badge-collection-edit-form/recipient-badge-collection-edit-form.component.ts

```diff
@@ -191,6 +191,14 @@ export class RecipientBadgeCollectionEditFormComponent extends BaseAuthenticated
 	}
 
 	badgeIssueDate(badge: BadgeResult) {
+		const context = badge.badge.apiModel.json['@context'];
+		const badgeVersion =
+			Array.isArray(context) && context.some((c) => c.indexOf('purl.imsglobal.org/spec/ob/v3p0') != -1)
+				? '3.0'
+				: '2.0';
+		if (badgeVersion == '3.0') {
+			return new Date(badge.badge.apiModel.json.validFrom);
+		}
```

Comment:
I think in this case, the model must be extended if we want to use it the same way as above.
Generally speaking though, it is safe to fallback between `validFrom` and `issuedOn` since they are exclusive to each other. No need to check explicitly for the version (which may come with other problems in the future).